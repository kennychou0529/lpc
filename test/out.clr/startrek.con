
compiling pass 1 (LPC)
-----------------------------------------------------
  warning on line 425: 'xpos' hiding existing declaration on line 390
  warning on line 425: 'ypos' hiding existing declaration on line 390
  warning on line 425: 'xinc' hiding existing declaration on line 390
  warning on line 425: 'yinc' hiding existing declaration on line 390
  warning on line 426: 'course' hiding existing declaration on line 389
  warning on line 426: 'warp' hiding existing declaration on line 390
  warning on line 539: 'i' hiding existing declaration on line 534
  warning on line 565: 'i' hiding existing declaration on line 534
  warning on line 665: non-local goto

0 error(s), 9 warning(s)

Exit code: 0

compiling pass 2 (target = clr)
-----------------------------------------------------

Invoking ILASM ...
Exit code: 0

Verifying the compiled binary (PEVERIFY) ...
[IL]: Error: [startrek.exe : <Module>::Printdigit][offset 0x0000000A][found address of Byte][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Printdigit][offset 0x00000022][found unmanaged pointer] Expected ByRef on the stack.
[IL]: Error: [startrek.exe : <Module>::Setupquad][offset 0x0000000B][found address of value 'T_sectxy'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Setupquad][offset 0x000000BB][found unmanaged pointer][expected readonly address of value 'T_sectxy'] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Setupquad][offset 0x000000D5][found unmanaged pointer][expected readonly address of value 'T_sectxy'] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Setupquad][offset 0x000000E5][found unmanaged pointer][expected readonly address of value 'T_sectxy'] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Setupquad][offset 0x000000F5][found unmanaged pointer][expected readonly address of value 'T_sectxy'] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Moveenterprise][offset 0x00000071][found address of Int32][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Moveenterprise][offset 0x00000098][found address of Double][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Firephasers][offset 0x0000004F][found address of Int32][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Firetorpedoes][offset 0x00000066][found address of Int32][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Selfdestruct][offset 0x0000005F][found address of Byte][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Command][offset 0x0000004F][found address of Byte][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::Instructions][offset 0x000001BF][found address of Byte][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::_SetContains][offset 0x00000004][found address of value 'T_set_27'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : <Module>::_SetContains][offset 0x00000005][found address of value 'T_set_28'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : T_array_1::arrayToString][offset 0x00000006][found address of value 'T_array_1'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : T_array_12::arrayToString][offset 0x00000006][found address of value 'T_array_12'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : T_klingonattack_array_23::arrayToString][offset 0x00000005][found address of value 'T_klingonattack_array_23'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : T_set_27::clear][offset 0x00000003] Instruction cannot be verified.
[IL]: Error: [startrek.exe : T_set_28::clear][offset 0x00000004] Instruction cannot be verified.
[IL]: Error: [startrek.exe : @moveenterprise::Moveintra][offset 0x00000021][found address of Double][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : @moveenterprise::Moveintra][offset 0x0000002B][found address of Double][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : @moveenterprise::Moveintra][offset 0x00000037][found address of Double][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : @moveenterprise::Moveintra][offset 0x00000041][found address of Double][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : @moveenterprise::Moveintra][offset 0x0000006F][found unmanaged pointer] Expected ByRef on the stack.
[IL]: Error: [startrek.exe : @firetorpedoes::Hitnova][offset 0x00000012][found address of Byte][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : @firetorpedoes::Hitnova][offset 0x000001D0][found unmanaged pointer] Expected ByRef on the stack.
[IL]: Error: [startrek.exe : @firetorpedoes::Hitklingbase][offset 0x00000012][found address of Byte][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [startrek.exe : @firetorpedoes::Hitklingbase][offset 0x00000043][found unmanaged pointer] Expected ByRef on the stack.
30 Error(s) Verifying startrek.exe
Exit code: 1

running the executable
-----------------------------------------------------

Exit code: 0