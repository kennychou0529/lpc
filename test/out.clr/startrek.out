Orders:  Stardate 3135

As commander of the United Starship Enterprise,
your mission is to rid the galaxy of the deadly
Klingon menace.  To do this, you must destroy the 
Klingon invasion force of 27 battle cruisers.
You have 27 solar years to complete
your mission (i.e. until stardate 3161).  The 
Enterprise is currently in quadrant 6-3, sector 2-4.

Do you need further instructions (y/n) ? 

The galaxy is divided into 64 quadrants with the
following coordinates:

  0   1   2   3   4   5   6   7
---------------------------------
|   |   |   |   |   |   |   |   |  0
---------------------------------
|   |   |   |   |   |   |   |   |  1
---------------------------------
|   |   |   |   |   |   |   |   |  2
---------------------------------
|   |   |   |   |   |   |   |   |  3
---------------------------------
|   |   |   |   |   |   |   |   |  4
---------------------------------
|   |   |   |   |   |   |   |   |  5
---------------------------------
|   |   |   |   |   |   |   |   |  6
---------------------------------
|   |   |   |   |   |   |   |   |  7

Each quadrant is similarly divided into 64 sectors.

Hit <return> to continue

::: DEVICES :::

 :: Warp Engines ::

 Course = a number in degrees.
   Numbers indicate direction starting at the top and
   going counter clockwise.

                     0
                 315 | 45
                    \\|/
               270 --*-- 90
                    /|\\
                 225 | 135
                    180

 Warp Factor = a REAL number from 0 to 12.
   Distance traveled = <warp factor> quadrants.
     Warp  .2  =  The Enterprise travels 1 sector.
           .5  =                         4 sectors.
            1  =                         1 quadrant.
            2  =                         2 quadrants.

Hit <return> to continue

   For example, if you travel from quadrant 1-1 in the
   direction of 90 degrees at warp 2, you would stop at
   quadrant 1-3 in the next stardate.  NOTE: every use of
   the warp engines takes one stardate.  If the Enterprise
   is blocked by something during an intra-quadrant travel,
   it will stop in front of it (and waste a stardate).

 :: Short Range Sensors ::

  The short range sensors display a detailed view of the 
  quadrant currently occupied by the Enterprise.  The 
  The following symbols have meanings as follows:

          Symbol          Meaning
             .            empty space
             *            a stable star
             +            an unstable star
             E            the Enterprise
             B            a Federation base
             K            a Klingon ship
             @            a Klingon base

Hit <return> to continue

 :: Long Range Sensors ::

  The long range sensors display the objects in the nine
  closest quadrants.  Each digit in each box means 
  means something:

    The ONES digit represents the number of STARS.
        TENS                                FEDERATION BASES.
        HUNDREDS                            KLINGON SHIPS
        THOUSANDS                           KLINGON BASES

  For example:
    319 means 3 Klingons, 1 Federation base, and 9 stars.
    206 means 2 Klingons, 0 Federation bases, and 6 stars.
   1007 means 1 Klingon base and 7 stars.

 :: Phasers ::

  Any portion of the energy available can be fired.  The
  battle computer divides this amount among the Klingon
  ships in the quadrant and determines the various directions

Hit <return> to continue

  of fire.  The effectiveness of a hit depends mostly on the
  distance to the target.  A Klingon battle cruiser starts with
  300 units of energy.  It can fire an amount equal to
  whatever energy is left.  Note that phasers are ineffective 
  against stars, Klingon bases, and Federation bases.

 :: Photon Torpedoes ::

  Initially the Enterprise has 10 photon torpedoes.
  One torpedo destroys whatever it hits.  The range of the
  photon torpedoes (like phasers) is limited to the current
  quadrant.  The course of a photon torpedo is set the same
  way as that of the Enterprise.  Torpedoes and phasers are
  restocked when the Enterprise docks at a Federation base.

 :: Damage Control Report ::

  The damage control report lists the state of repair of each
  device.  A non-zero state indicates the number of stardates
  required to repair the device.  Devices can be damaged by
  space storms, and are repaired by time and truces.

Hit <return> to continue

 :: History Computers ::

  The history computers keep a record of all the quadrants
  scanned with the long range scanners.  The history report
  uses the same display format as the long range scanners,
  except that the entire galaxy is displayed.  A quadrant
  that has not been scanned is printed as "+++++".

 :: Suicide Device ::

  It is possible to implement a self-destruct sequence merely
  by invoking this command.  The game is terminated.

To get a list of all commands, type "9" when asked for a
command.  All commands are terminated by the [RETURN] key.
You have at least on supporting starbase.  Your energy and
photon torpedoes are replenished when you are docked at a
Federation starbase.  G O O D   L U C K !


Hit <return> to continue


----------------------
. . . . . . . .    Stardate         3135
. . . . . . . .    Condition        Green     
. . . . E . . .    Quadrant         6-3
. . . . . . . .    Sector           2-4
. . . . . . . .    Energy           5000
. . . . . . . .    Photon torpedoes 10
. . . . . . . .    Klingons left    27
. . . . . . . .    
----------------------
Command: 
0 = Set course
1 = Short range sensor scan
2 = Long range sensor scan
3 = Fire phasors
4 = Fire photon torpedoes
5 = Damage control report
6 = History computer report
7 = Self destruct
Command: 
----------------------
. . . . . . . .    Stardate         3135
. . . . . . . .    Condition        Green     
. . . . E . . .    Quadrant         6-3
. . . . . . . .    Sector           2-4
. . . . . . . .    Energy           5000
. . . . . . . .    Photon torpedoes 10
. . . . . . . .    Klingons left    27
. . . . . . . .    
----------------------
Command: 
Long Range Sensor Scan For Quadrant 6-3
|-----|-----|-----|
|   5 |   0 |   1 |
|-----|-----|-----|
|   1 |   0 | 100 |
|-----|-----|-----|
|1105 |   5 | 105 |
|-----|-----|-----|
Command: 
History Computer Report; Stardate 3135
|-----|-----|-----|-----|-----|-----|-----|-----|
|+++++|+++++|+++++|+++++|+++++|+++++|+++++|+++++|
|-----|-----|-----|-----|-----|-----|-----|-----|
|+++++|+++++|+++++|+++++|+++++|+++++|+++++|+++++|
|-----|-----|-----|-----|-----|-----|-----|-----|
|+++++|+++++|+++++|+++++|+++++|+++++|+++++|+++++|
|-----|-----|-----|-----|-----|-----|-----|-----|
|+++++|+++++|+++++|+++++|+++++|+++++|+++++|+++++|
|-----|-----|-----|-----|-----|-----|-----|-----|
|+++++|+++++|+++++|+++++|+++++|+++++|+++++|+++++|
|-----|-----|-----|-----|-----|-----|-----|-----|
|+++++|+++++|   5 |   0 |   1 |+++++|+++++|+++++|
|-----|-----|-----|-----|-----|-----|-----|-----|
|+++++|+++++|   1 |   0 | 100 |+++++|+++++|+++++|
|-----|-----|-----|-----|-----|-----|-----|-----|
|+++++|+++++|1105 |   5 | 105 |+++++|+++++|+++++|
|-----|-----|-----|-----|-----|-----|-----|-----|
Command: 
Course: Warp factor (0-12): 97 unit hit on Enterprise from Klingon at sector 3-7 (4723 left)
150 unit hit on Enterprise from Klingon at sector 4-5 (4573 left)
112 unit hit on Enterprise from Klingon at sector 5-5 (4461 left)
102 unit hit on Enterprise from Klingon at sector 5-7 (4359 left)
----------------------
. . . . . . . .    Stardate         3136
. . . . . . . .    Condition        Red       
* . * E . . . .    Quadrant         0-4
. . . * . . . K    Sector           2-3
. . . . . K . .    Energy           4359
. . . . . K . K    Photon torpedoes 10
. . . . . . . .    Klingons left    27
. . . . . . . .    
----------------------
Command: 
Phasers locked on target.  Energy available = 4359
Number of units to fire: 17 unit hit on Klingon at sector 3-7 (283 left)
26 unit hit on Klingon at sector 4-5 (274 left)
25 unit hit on Klingon at sector 5-5 (275 left)
11 unit hit on Klingon at sector 5-7 (289 left)
120 unit hit on Enterprise from Klingon at sector 3-7 (4189 left)
118 unit hit on Enterprise from Klingon at sector 4-5 (4071 left)
103 unit hit on Enterprise from Klingon at sector 5-5 (3968 left)
86 unit hit on Enterprise from Klingon at sector 5-7 (3882 left)
----------------------
. . . . . . . .    Stardate         3136
. . . . . . . .    Condition        Red       
* . * E . . . .    Quadrant         0-4
. . . * . . . K    Sector           2-3
. . . . . K . .    Energy           3882
. . . . . K . K    Photon torpedoes 10
. . . . . . . .    Klingons left    27
. . . . . . . .    
----------------------
Command: 
Phasers locked on target.  Energy available = 3882
Number of units to fire: 23 unit hit on Klingon at sector 3-7 (260 left)
16 unit hit on Klingon at sector 4-5 (258 left)
16 unit hit on Klingon at sector 5-5 (259 left)
17 unit hit on Klingon at sector 5-7 (272 left)
104 unit hit on Enterprise from Klingon at sector 3-7 (3728 left)
172 unit hit on Enterprise from Klingon at sector 4-5 (3556 left)
77 unit hit on Enterprise from Klingon at sector 5-5 (3479 left)
108 unit hit on Enterprise from Klingon at sector 5-7 (3371 left)
----------------------
. . . . . . . .    Stardate         3136
. . . . . . . .    Condition        Red       
* . * E . . . .    Quadrant         0-4
. . . * . . . K    Sector           2-3
. . . . . K . .    Energy           3371
. . . . . K . K    Photon torpedoes 10
. . . . . . . .    Klingons left    27
. . . . . . . .    
----------------------
Command: 
Torpedo course: Torpedo track:
2-3
1-3
0-3
Torpedo missed.
78 unit hit on Enterprise from Klingon at sector 3-7 (3293 left)
163 unit hit on Enterprise from Klingon at sector 4-5 (3130 left)
110 unit hit on Enterprise from Klingon at sector 5-5 (3020 left)
81 unit hit on Enterprise from Klingon at sector 5-7 (2939 left)
----------------------
. . . . . . . .    Stardate         3136
. . . . . . . .    Condition        Red       
* . * E . . . .    Quadrant         0-4
. . . * . . . K    Sector           2-3
. . . . . K . .    Energy           2939
. . . . . K . K    Photon torpedoes 9
. . . . . . . .    Klingons left    27
. . . . . . . .    
----------------------
Command: 
Device name:      Repair Time:
Warp Engines            0
Short Range Sensors     0
Long Range Sensors      0
Phaser Control          0
Photon Tubes            0
Damage Control          0
History Computers       0
Self Destruct           0
Command: 
Are you SURE ? 
